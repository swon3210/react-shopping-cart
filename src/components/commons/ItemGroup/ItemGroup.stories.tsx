import styled from 'styled-components';
import ItemGroup, { Props } from './ItemGroup';
import ListItem from '../ListItem/ListItem';

export default {
  title: 'Components/Commons/ItemGroup',
  component: ItemGroup,
  argTypes: {},
};

const ItemWrapper = styled.div`
  padding: 20px;
`;

const Template = (args: Props) => (
  <ItemGroup {...args}>
    <ItemWrapper>
      <ListItem
        thumbnail={
          'https://images.unsplash.com/photo-1589365278144-c9e705f843ba?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=634&q=80'
        }
        name={'상품 이름'}
        price={'10,000'}
        quantity={'3'}
        size={'SM'}
      />
    </ItemWrapper>
  </ItemGroup>
);

export const Default = Template.bind({});

(Default as any).args = {
  orderNumber: '2',
};
