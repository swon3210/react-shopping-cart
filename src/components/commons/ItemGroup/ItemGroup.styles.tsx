import styled from 'styled-components';

export const ItemGroup = styled.div`
  border: 1px solid ${({ theme }) => theme.GRAY_400};
`;

export const Header = styled.div`
  display: flex;
  justify-content: space-between;
  background-color: ${({ theme }) => theme.GRAY_100};
  padding: 36px 39px;
  border-bottom: 2px solid ${({ theme }) => theme.GRAY_400};
  color: ${({ theme }) => theme.TEXT_COLOR};
  font-size: 20px;
  letter-spacing: 0.5px;
  line-height: 24px;
`;

export const HeaderText = styled.span``;

export const HeaderLink = styled.button``;

export const ItemContainer = styled.div``;
