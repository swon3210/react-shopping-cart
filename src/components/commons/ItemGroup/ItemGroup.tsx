import * as Styled from './ItemGroup.styles';

export interface Props {
  children: React.ReactNode;
  headerText: string;
  headerLinkButtonText?: string;
  onHeaderLinkClick?: () => void;
}

const ItemGroup = ({ children, headerText, onHeaderLinkClick, headerLinkButtonText }: Props) => {
  return (
    <Styled.ItemGroup>
      <Styled.Header>
        <Styled.HeaderText>주문번호 : {headerText}</Styled.HeaderText>
        {headerLinkButtonText && (
          <Styled.HeaderLink onClick={onHeaderLinkClick}>{headerLinkButtonText}</Styled.HeaderLink>
        )}
      </Styled.Header>
      <Styled.ItemContainer>{children}</Styled.ItemContainer>
    </Styled.ItemGroup>
  );
};

export default ItemGroup;
