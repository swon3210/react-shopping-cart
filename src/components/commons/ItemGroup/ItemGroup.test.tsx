import ItemGroup from './ItemGroup';
import { render } from '@testing-library/react';

describe('Checkbox Component', () => {
  it('Checkbox Snapshot', () => {
    const itemGroupboxUtil = render(<ItemGroup>TEST</ItemGroup>);
    expect(itemGroupboxUtil.asFragment()).toMatchSnapshot();
  });
});
