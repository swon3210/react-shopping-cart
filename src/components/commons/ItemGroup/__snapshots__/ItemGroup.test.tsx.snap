// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component Checkbox Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa jBmzxu"
  >
    <div
      class="sc-gsDJrp kyLDcO"
    >
      <span
        class="sc-dkPtyc"
      >
        주문번호 : 
      </span>
    </div>
    <div
      class="sc-eCImvq"
    >
      TEST
    </div>
  </div>
</DocumentFragment>
`;
